apisix/consumer.lua

```diff
@@ -88,6 +88,11 @@ function _M.attach_consumer(ctx, consumer, conf)
     ctx.consumer_ver = conf.conf_version
 end
 
+-- attach chosen consumer to the request headers, used in auth plugin
+function _M.attach_consumer_to_request(ctx, req, consumer)
+    req.set_header(ctx, "X-Consumer-Name", consumer.consumer_name)
+    req.set_header(ctx, "X-Consumer-Group-ID", consumer.group_id)
```

Comment:
Can't the user use proxy-rewrite to add them?
https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/proxy-rewrite.md
We support using variables in the header and we have these variables: https://github.com/apache/apisix/blob/master/docs/en/latest/apisix-variable.md

Reply:
Yes. Of course, users can use proxy-rewrite to implement it, but if APISIX supports adding the name directly to the header, the difference is more convenient.

Reply:
Not all users think that they can go through proxy-rewrite to add the consumer name to the header. This method is too tricky